modules/scheduler-impl/src/main/java/org/opencastproject/scheduler/endpoint/SchedulerRestService.java

```diff
@@ -1084,57 +1095,64 @@ public Response getEventsAsXml(@QueryParam("agent") String device, @QueryParam("
-                  @RestResponse(responseCode = HttpServletResponse.SC_BAD_REQUEST, description = "Missing or invalid parameters") })
-  public Response getConflictingEventsXml(@QueryParam("agent") String device, @QueryParam("rrule") String rrule,
+                  @RestResponse(responseCode = HttpServletResponse.SC_BAD_REQUEST, description = "Missing or invalid parameters"),
+                  @RestResponse(responseCode = HttpServletResponse.SC_UNAUTHORIZED, description = "Not authorized to make this request"),
+                  @RestResponse(responseCode = HttpServletResponse.SC_INTERNAL_SERVER_ERROR, description = "A detailed stack track of the internal issue.")})
+  public Response getConflictingEventsJson(@QueryParam("agent") String device, @QueryParam("rrule") String rrule,
           @QueryParam("start") Long startDate, @QueryParam("end") Long endDate, @QueryParam("duration") Long duration,
           @QueryParam("timezone") String timezone) throws UnauthorizedException {
-    if (StringUtils.isBlank(device) || startDate == null || endDate == null) {
-      logger.info("Either agent, start date or end date were not specified");
-      return Response.status(Status.BAD_REQUEST).build();
-    }
-
-    RRule rule = null;
-    if (StringUtils.isNotBlank(rrule)) {
-      if (duration == null || StringUtils.isBlank(timezone)) {
-        logger.info("Either duration or timezone were not specified");
-        return Response.status(Status.BAD_REQUEST).build();
-      }
-
-      try {
-        rule = new RRule(rrule);
-        rule.validate();
-      } catch (Exception e) {
-        logger.info("Unable to parse rrule {}: {}", rrule, getMessage(e));
-        return Response.status(Status.BAD_REQUEST).build();
-      }
-
-      if (!Arrays.asList(TimeZone.getAvailableIDs()).contains(timezone)) {
-        logger.info("Unable to parse timezone: {}", timezone);
-        return Response.status(Status.BAD_REQUEST).build();
+    try {
+      List<MediaPackage> events = getConflictingEvents(device, rrule, startDate, endDate, duration, timezone);
+      if (!events.isEmpty()) {
+        String eventsJsonString = getEventListAsJsonString(events);
+        return Response.ok(eventsJsonString).build();
+      } else {
+        return Response.noContent().build();
       }
+    } catch (IllegalArgumentException e) {
+      return Response.status(Status.BAD_REQUEST).build();
+    } catch (UnauthorizedException e) {
+      throw e;
+    } catch (Exception e) {
+      logger.error("Unable to find conflicting events for {}, {}, {}, {}, {}: {}",
+              device, rrule, startDate, endDate, duration, getStackTrace(e));
+      throw new WebApplicationException(Response.Status.INTERNAL_SERVER_ERROR);
     }
+  }
 
-    Date start = new DateTime(startDate).toDateTime(DateTimeZone.UTC).toDate();
-
-    Date end = new DateTime(endDate).toDateTime(DateTimeZone.UTC).toDate();
+  @GET
+  @Produces(MediaType.TEXT_XML)
+  @Path("conflicts.xml")
```

Comment:
Hmm... would you be willing to also do a  @Path("recordings.{type:xml|json}") here (instead of adding a second method)? Seems that avoid all the overhead of the REST docs...

Reply:
+1